aviary/interface/methods_for_level2.py

```diff
@@ -1394,13 +1486,24 @@ def add_linkage_constraint(self, phase_a, phase_b, var_b):
             self.traj.link_phases(phases[:7], vars=["TAS"], units='kn', ref=200.)
 
         elif self.mission_method is HEIGHT_ENERGY:
+
+            # connect mass and distance between all phases regardless of reserve / non-reserve status
             self.traj.link_phases(
                 phases, ["time", Dynamic.Mission.MASS, Dynamic.Mission.DISTANCE], connected=True)
 
+            regular_phases, reserve_phases = self.phase_separator()
+
+            # connect regular phases with eachother if your are optimizing alt or mach
+            self._link_phases_helper_with_options(
+                regular_phases, 'optimize_altitude', Dynamic.Mission.ALTITUDE, ref=1.e4)
+            self._link_phases_helper_with_options(
+                regular_phases, 'optimize_mach', Dynamic.Mission.MACH)
+
+            # connect reserve phases with eachother if your are optimizing alt or mach
```

Comment:
```suggestion
            # connect reserve phases with each other if you're are optimizing alt or mach
```

Reply:
it looks like these are happening every time, not just when optimizing alt or mach?

Reply:
`_link_phases_helper_with_options()` doesn't connect anything if the phase dictionaries are empty.